### PR TITLE
Feature 106 support for changing layout element background colors per page

### LINKED ISSUE #106: Support for changing layout element background colors per page

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #106

### CODE REVIEW

File: src/components/screens/content-editor-screen.tsx [hunk 1]
```diff
@@ -1791,6 +1799,17 @@ class ContentEditorScreen extends React.Component<Props, State> {
     return layouts.find((layout) => layout.id === page.layoutId) || null;
   };
 
+  private validatePageResource = (
```
Comment (anttileppa) [inline src/components/screens/content-editor-screen.tsx:1802]: Add ts docs

File: src/components/screens/layout-screen-html.tsx [hunk 2]
```diff
@@ -196,8 +207,7 @@ const LayoutScreenHTML: FC<Props> = ({
     },
     {
       name: strings.exhibitionLayouts.editView.saveButton,
-      action: onLayoutSave,
-      disabled: !dataChanged
```
Comment (anttileppa) [inline src/components/screens/layout-screen-html.tsx:200]: Why data changed functionality has been removed?

File: src/utils/generic-utils.ts [hunk 3]
```diff
@@ -77,6 +77,32 @@ namespace GenericUtils {
...
+  /**
+   * Converts CSS HEX color to CSS RGB function
+   *
+   * If the input is already in RGB format, it will be returned as is.
+   * If the input is not a valid HEX, undefined will be returned.
+   *
+   * @param hexColor hex color string
+   * @returns rgb color string
+   */
+  export const hexToRGB = (hexColor: string) => {
```
Comment (anttileppa) [inline src/utils/generic-utils.ts:90]: Maybe you should use some kind of color library to abstract parsing e.g. https://www.npmjs.com/package/color-parse

File: src/components/content-editor/resource-editor.tsx [hunk 4]
```diff
@@ -172,6 +174,15 @@ class ResourceEditor extends React.Component<Props, {}> {
             onChange={this.onResourceDataChange}
           />
         );
+      case ExhibitionPageResourceType.Color:
+        return (
+          <ColorPicker
+            color={resource.data}
+            onChangeComplete={({ rgb: { r, g, b, a } }) =>
+              this.updateResourceData(`rgba(${r}, ${g}, ${b}, ${a})`)
```
Comment (anttileppa) [inline src/components/content-editor/resource-editor.tsx:182]: Check whether alpha can be undefined, if it can then default to 100%

**Review by anttileppa — APPROVED** [review]

File: src/components/screens/content-editor-screen.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/screens/content-editor-screen.tsx:1809]: Resolved.

File: src/components/screens/layout-screen-html.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/screens/layout-screen-html.tsx]: Added it back per discussions

File: src/utils/generic-utils.ts (hunk 3 above)
Comment (villejuutila) [reply src/utils/generic-utils.ts:91]: Resolved

File: src/components/content-editor/resource-editor.tsx (hunk 4 above)
Comment (villejuutila) [reply src/components/content-editor/resource-editor.tsx]: Resolved.

**Pull request merged**